x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/FillMaskConfigUpdate.java

```diff
@@ -0,0 +1,179 @@
+
+    private static ObjectParser<FillMaskConfigUpdate.Builder, Void> createParser(boolean lenient) {
+        ObjectParser<FillMaskConfigUpdate.Builder, Void> parser = new ObjectParser<>(
+            NAME,
+            lenient,
+            FillMaskConfigUpdate.Builder::new);
+        parser.declareString(FillMaskConfigUpdate.Builder::setResultsField, RESULTS_FIELD);
+        parser.declareInt(FillMaskConfigUpdate.Builder::setNumTopClasses, NUM_TOP_CLASSES);
+        return parser;
+    }
+
+    public static FillMaskConfigUpdate fromXContentStrict(XContentParser parser) {
+        return STRICT_PARSER.apply(parser, null).build();
+    }
+
+    private final Integer numTopClasses;
+    private final String resultsField;
+
+    public FillMaskConfigUpdate(Integer numTopClasses, String resultsField) {
+        this.numTopClasses = numTopClasses;
+        this.resultsField = resultsField;
+    }
+
+    public FillMaskConfigUpdate(StreamInput in) throws IOException {
+        this.numTopClasses = in.readOptionalVInt();
+        this.resultsField = in.readOptionalString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalVInt(numTopClasses);
+        out.writeOptionalString(resultsField);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (numTopClasses != null) {
+            builder.field(NUM_TOP_CLASSES.getPreferredName(), numTopClasses);
+        }
+        if (resultsField != null) {
+            builder.field(RESULTS_FIELD.getPreferredName(), resultsField);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public InferenceConfig apply(InferenceConfig originalConfig) {
+        if (originalConfig instanceof FillMaskConfig == false) {
+            throw ExceptionsHelper.badRequestException(
+                "Inference config of type [{}] can not be updated with a request of type [{}]",
+                originalConfig.getName(),
+                getName());
+        }
+
+        FillMaskConfig fillMaskConfig = (FillMaskConfig)originalConfig;
+        if (isNoop(fillMaskConfig)) {
+            return originalConfig;
+        }
+
+        FillMaskConfig.Builder builder = new FillMaskConfig.Builder(fillMaskConfig);
+        if (numTopClasses != null) {
+            builder.setNumTopClasses(numTopClasses);
+        }
+        if (resultsField != null) {
+            builder.setResultsField(resultsField);
+        }
+        return builder.build();
+    }
+
+    boolean isNoop(FillMaskConfig originalConfig) {
+        return (this.numTopClasses == null || this.numTopClasses == originalConfig.getNumTopClasses()) &&
+            (this.resultsField == null);
```

Comment:
```suggestion
            (this.resultsField == null || this.resultsField.equals(originalConfig.getResultsField()));
```
Obviously, requires a getter for `results_field`